cstore_writer.c

```diff
@@ -191,12 +174,37 @@ CStoreBeginWrite(const char *filename, CompressionType compressionType,
 	writeState->stripeWriteContext = stripeWriteContext;
 	writeState->blockDataArray = blockData;
 	writeState->compressionBuffer = NULL;
+	writeState->activeBlockNumber = blockNumber;
+	writeState->relation = relation;
 
 	return writeState;
 }
 
 
 /*
+ * EnsureDataForkExists checks if data fork exists for writing data,
+ * and creates if it is not present.
+ */
+static void EnsureDataForkExists(Relation relation, bool logging)
+{
+	SMgrRelation srel = smgropen(relation->rd_node, InvalidBackendId);
+
+	if (!smgrexists(srel, DATA_FORKNUM))
+	{
+		smgrcreate(srel, DATA_FORKNUM, false);
+
+		if (logging)
+		{
+			log_smgrcreate(&srel->smgr_rnode.node, DATA_FORKNUM);
```

Comment:
Why are passing through a `logging` option here? Couldn't we just use `RelationNeedsWAL()`?